pkg/apis/serving/k8s_validation.go

```diff
@@ -267,7 +277,60 @@ func ValidatePodSpec(ps corev1.PodSpec) *apis.FieldError {
 	return errs
 }
 
+//ValidateMultiContainerPorts validates port when specified multiple containers
+func ValidateMultiContainerPorts(containers []corev1.Container) *apis.FieldError {
+	var (
+		count int
+		errs  *apis.FieldError
+	)
+	for i := range containers {
+		for range containers[i].Ports {
+			count++
+		}
+	}
+	if count == 0 {
+		errs = errs.Also(apis.ErrMissingField(apis.CurrentField))
+	}
+	return errs.Also(portValidation(count))
+}
+
+//ValidateSidecarContainer validate fields for non serving containers
+func ValidateSidecarContainer(container *corev1.Container, volumes sets.String) *apis.FieldError {
+	var errs *apis.FieldError
+	validateMultiContainerProbe := func(p *corev1.Probe) *apis.FieldError {
+		if p != nil {
+			return apis.CheckDisallowedFields(*p, *ProbeMask(&corev1.Probe{}))
+		}
+		return nil
+	}
+	errs = errs.Also(validateMultiContainerProbe(container.LivenessProbe).ViaField("livenessProbe"))
+	errs = errs.Also(validateMultiContainerProbe(container.ReadinessProbe).ViaField("readinessProbe"))
+	return errs.Also(validate(*container, volumes))
+}
+
+//ValidateContainer validate fields for serving containers
 func ValidateContainer(container corev1.Container, volumes sets.String) *apis.FieldError {
+	var errs *apis.FieldError
+	// Liveness Probes
+	errs = errs.Also(validateProbe(container.LivenessProbe).ViaField("livenessProbe"))
+	// Readiness Probes
+	errs = errs.Also(validateReadinessProbe(container.ReadinessProbe).ViaField("readinessProbe"))
+	return errs.Also(validate(container, volumes))
+}
+
+func portValidation(count int) *apis.FieldError {
+	var errs *apis.FieldError
+	if count > 1 {
+		errs = errs.Also(&apis.FieldError{
```

Comment:
```suggestion
		return errs.Also(&apis.FieldError{
```

You can then remove the `errs` variable and return nil explicitly below.